oioswift/common/middleware/copy.py

```diff
@@ -186,6 +187,54 @@ def _load_object_post_as_copy_conf(self, conf):
         except (NoSectionError, NoOptionError):
             pass
 
+    def allow_fast_copy(self, req):
+        """Check is fastcopy is possible."""
+
+        if req.method != 'PUT' or 'X-Copy-From' not in req.headers:
+            return False
+
+        # TODO maybe is it not necessary
+        already_checked = req.headers.get("Oio-Check-Fastcopy", 0)
+        if already_checked:
+            return False
```

Comment:
If the previous check returned `True`, you should return `True`.

Reply:
No, `Oio-Check-Fastcopy` indicate that check if `FastCopy` is possible or not, not that `FastCopy` is already done.
Some middleware can launch requests and we don't want to poll properties objects